File: OBI-oficial/2024/Fase 1/relogio.py
horas = int(input())
minutos = int(input())
segundos = int(input())

segundosAdiado = int(input())

# Convert all time to seconds
total_segundos = horas * 3600 + minutos * 60 + segundos
# Add the specified seconds
total_segundos += segundosAdiado

# Calculate the new time components
horas = (total_segundos // 3600) % 24
minutos = (total_segundos // 60) % 60
segundos = total_segundos % 60

print(horas)
print(minutos)
print(segundos)

File: Estudo-2025/cpp/2024/relogio.cpp
#include <iostream>
using namespace std;
int main()
{
    int horas, minutos, segundos, adiamento;
    cin >> horas >> minutos >> segundos >> adiamento;

    segundos += adiamento % 60;

    if (segundos >= 60)
    {
        segundos -= 60;
        minutos += 1;
    }
    adiamento = adiamento / 60;
    minutos += adiamento % 60;
    if (minutos >= 60)
    {
        minutos -= 60;
        horas += 1;
    }

    adiamento = adiamento / 60;
    horas += adiamento % 24;

    cout << horas << endl;
    cout << minutos << endl;
    cout << segundos << endl;

    return 0;
}

/*
3
14
15
1
 */